### ISSUE TITLE
Support textual diffs for docm and dotm

### ISSUE THREAD

**Issue by atavakoli** [git-for-windows/git#900]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen)
   or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue
   matching what I'm seeing

**Note:** #355 exists, but has been closed, and doesn't mention `do[ct]m` files. If preferred,
I can close this issue and request reopening that one.

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

64-bit

```
$ git --version --build-options
git version 2.7.0.windows.1
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

Windows 8.1 Pro 64-bit

```
$ cmd.exe /c ver
Microsoft Windows [Version 6.3.9600]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
$cat /etc/install-options.txt
Path Option: BashOnly
SSH Option: OpenSSH
CRLF Option: CRLFCommitAsIs
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

No, but I can get the desired behavior by updating /mingw64/gitattributes and
/usr/bin/astextplain like so (respectively):

```diff
@@ -2,8 +2,12 @@
 *.DOC  diff=astextplain
 *.docx diff=astextplain
 *.DOCX diff=astextplain
+*.docm diff=astextplain
+*.DOCM diff=astextplain
 *.dot  diff=astextplain
 *.DOT  diff=astextplain
+*.dotm diff=astextplain
+*.DOTM diff=astextplain
 *.pdf  diff=astextplain
 *.PDF  diff=astextplain
 *.rtf  diff=astextplain
```

```diff
@@ -11,7 +11,7 @@ case "$1" in
        *.doc | *.DOC | *.dot | *.DOT)
                antiword -m UTF-8 "$1" | sed "s/\^M$//" || cat "$1"
                ;;
-       *.docx | *.DOCX)
+       *.docx | *.DOCX | *.docm | *.DOCM | *.dotm | *.DOTM)
                docx2txt.pl "$1" -
                ;;
        *.pdf | *.PDF)
@@ -27,4 +27,4 @@ case "$1" in
                ;;
 esac
```

I have a branch with the above fixes (along with another for git-extra/git-extra.install), ready to be pushed to the git-for-windows/build-extra project.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

Bash

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
<create myfile.docm with some content>

$ git add myfile.docm

$ git commit -m "Add docm file"
[master f0f0f0f] Add docm file
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 myfile.docm

$ git show
commit f0f0f0f000000000000000000000000000000
Author: Me <me@me.me>
Date:   Fri Sep 23 16:44:33 2016 -0400

    Add docm file

diff --git a/myfile.docm b/myfile.docm
new file mode 100644
index 0000000..e8186b6
Binary files /dev/null and b/myfile.docm differ
```

 - What did you expect to occur after running these commands?

I expected to see the new file's contents after running `git show` (i.e. what I would've seen if I'd
saved the file as .docx).

 - What actually happened instead?

I got "Binary files ... differ" instead of a textual diff.

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

N/A

**Comment by dscho** [comment]
Could you open a PR for https://github.com/git-for-windows/build-extra/blob/master/git-extra/gitattributes and https://github.com/git-for-windows/build-extra/blob/master/git-extra/astextplain, please? Just have a look at the commit history to see the preferred commit message format.

**Comment by atavakoli** [comment]
@dscho Done (PR 128). I included the additional .install file, but can revert that change if necessary.

**Comment by dscho** [comment]
Thanks. I merged this PR (git-for-windows/build-extra#128), touched things up a bit and it will be part of the next version.
